File: frontend/src/types/history.ts
export interface RelatedPage {
  title: string;
  url: string;
}

export interface HistoricalEvent {
  label: string;
  date?: string;
  year?: number;
  description: string;
  wikidata_id?: string;
  wikipedia_url?: string;
  image_url?: string;
  category?: string;
  title?: string;
  related_pages?: RelatedPage[];
}

export interface OnThisDayEvent {
  year: number;
  title: string;
  description: string;
  wikipedia_url?: string;
  related_pages?: RelatedPage[];
}

export interface CountryHistory {
  country_name: string;
  iso_code: string;
  summary?: string;
  wikipedia_url?: string;
  events: HistoricalEvent[];
}

File: backend/app/schemas/history.py
from pydantic import BaseModel


class HistoricalEvent(BaseModel):
    label: str = ""
    date: str = ""
    year: int | None = None
    description: str = ""
    wikidata_id: str = ""
    wikipedia_url: str | None = None
    image_url: str | None = None


class CountryHistory(BaseModel):
    country_name: str
    iso_code: str
    events: list[HistoricalEvent] = []
